### PR TITLE
EventPipe lock implementation

### PR DESCRIPTION
**Pull request by LakshanF**
Uses CrstStatic as the locking mechanism for NativeAOT EventPipe.

### CODE REVIEW

**Comment by LakshanF** [comment]
The extra-platform NativeAOT [failures ](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/188611/logs/730)(win-x64) seems unrelated with large number of [Cryptography](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-82790-merge-06efc20c2dab4d9e8b/System.Security.Cryptography.Cng.Tests/1/console.c87ec695.log?helixlogtype=result) failures and some [Http](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-82790-merge-06efc20c2dab4d9e8b/System.Net.Http.WinHttpHandler.Functional.Tests/3/console.99f3087c.log?helixlogtype=result) functional ones.

File: src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp [hunk 1]
```diff
@@ -502,4 +509,77 @@ ep_rt_aot_volatile_store_ptr_without_barrier (
     VolatileStoreWithoutBarrier<void *> ((void **)ptr, value);
 }
 
+void ep_rt_aot_init (void)
+{
+    extern ep_rt_lock_handle_t _ep_rt_aot_config_lock_handle;
+    extern CrstStatic _ep_rt_aot_config_lock;
+
+    _ep_rt_aot_config_lock_handle.lock = &_ep_rt_aot_config_lock;
+    _ep_rt_aot_config_lock_handle.lock->InitNoThrow (CrstType::CrstEventPipe);
```
Comment (VSadov) [inline src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp:518]: `CrstType` is unused in NativeAOT, but good for accounting where all the locks are. I think this should use a separate type, something like `CrstType::CrstEventPipeConfig`

**Review by VSadov — APPROVED** [review]
LGTM!

**Comment by VSadov** [comment]
New changes look good. :shipit:

**Comment by MichalStrehovsky** [comment]
> No hay más extremos disponibles desde el asignador de extremos

This is #74838 and unrelated (assuming all the failures are this - didn't check - one of the problems with noisy failure modes like this is that it's easy to miss the one relevant regression if there is one).

**Pull request merged**